### PR TITLE
feat: replace seaborn with matplotlib for box_plot

### LINKED ISSUE #805: Consistent box plots for tables and columns
### Is your feature request related to a problem?

`Table.plot.box_plots` and `Column.plot.box_plot` produce differently styled box plots.

### Desired solution

They should be consistent and as readable as possible.

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### LINKED ISSUE #849: `plot.box_plots` with matplotlib
### Is your feature request related to a problem?

`plot.box_plots` really is the last plot method that still uses seaborn. It should be rewritten using matplotlib, so we can remove the dependency on seaborn.

### Desired solution

/

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by Muellersen**
Closes #805
Closes #849 

### Summary of Changes

This pull request applies the changes of replacing the last import of seaborn with matplotlib, so that seaborn will be removed as dependency with this pull request.

We also replaced the snapshot images of the box_plot test, to ensure that the test run successfully.

### CODE REVIEW

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 1]
```diff
@@ -34,48 +34,36 @@ def __init__(self, table: Table):
 
     def box_plots(self) -> Image:
         """
-        Plot a boxplot for every numerical column.
+        Create a box plot for the values in the column. This is only possible for numeric columns.
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:37]: The previous documentation fit better. You say "This is only possible for numeric columns.", so as a user I'd wonder how I could control for which columns box plots should be created. However, this decision happens automatically.

```suggestion
        Create a box plot for every numerical column.
```

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 2]
```diff
@@ -34,48 +34,36 @@ def __init__(self, table: Table):
...
+            If the column is not numeric.
 
         Examples
         --------
-        >>> from safeds.data.tabular.containers import Table
-        >>> table = Table({"a":[1, 2], "b": [3, 42]})
-        >>> image = table.plot.box_plots()
+        >>> from safeds.data.tabular.containers import Column
+        >>> column = Column("test", [1, 2, 3])
+        >>> boxplot = column.plot.box_plot()
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:53]: We're working on a table, not a column.

Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py:53]: We're working on a table, not a column here.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: src/safeds/data/tabular/plotting/_table_plotter.py (hunk 2 above)
Comment (Muellersen) [reply src/safeds/data/tabular/plotting/_table_plotter.py]: This was a mistake and is fixed

File: src/safeds/data/tabular/plotting/_table_plotter.py (hunk 1 above)
Comment (Muellersen) [reply src/safeds/data/tabular/plotting/_table_plotter.py]: We reverted the change

File: docs/tutorials/data_visualization.ipynb
Comment (lars-reimann) [inline docs/tutorials/data_visualization.ipynb:1]: Where did the box plots for `travel_class`, `fare`, and `survived` go?

**Review by lars-reimann — CHANGES_REQUESTED** [review]

File: docs/tutorials/data_visualization.ipynb
Comment (Muellersen) [reply docs/tutorials/data_visualization.ipynb:1]: Fixed

File: src/safeds/data/tabular/plotting/_table_plotter.py [hunk 3]
```diff
@@ -89,13 +96,11 @@ def box_plots(self) -> Image:
...
         for i in range(last_filled_ax_index, number_of_columns):
-            if number_of_rows == 1:
-                fig.delaxes(axs[i])
-            else:
+            if number_of_rows != 1 and last_filled_ax_index != 0:
                 fig.delaxes(axs[number_of_rows - 1, i])
 
         fig.tight_layout()
-
+        _figure_to_image(fig).to_png_file("C:/Users/patri/Desktop/test.png")
```
Comment (lars-reimann) [inline src/safeds/data/tabular/plotting/_table_plotter.py]: Absolute paths should not be part of the repo.

To update snapshots, run tests using `pytest --snapshot-update`. `syrupy` automatically takes care of snapshot updates for you. You can also run tests only partially with that flag to speed up the process.

**Review by lars-reimann — CHANGES_REQUESTED** [review]

**Review by lars-reimann — APPROVED** [review]
Nicely done, thanks!

**Pull request merged**